models/collection_error_model.py

```
@@ -0,0 +1,30 @@
+
+    def __init__(self, data):
+        for key, item in data.items():
+            setattr(self, key, item)
+    
+    def save(self):   
+        with current_app.app_context():
+            try:
+                db.session.add(self)
+                db.session.commit()
+            except SQLAlchemyError as err:
+                raise DatabaseError("Database Error", err)
```

Ruff (0.13.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)